### PR TITLE
ja: : cr2 Day 2 translations

### PR DESCRIPTION
**Pull request by henrif75**
Professional translations for CR2 content.
Changes are being staged in a separate branch until all Japanese content is reviewed.

#1463 #652

### CODE REVIEW

**Comment by kantasv** [comment]
Thanks for raising the MR. I'll take a look at this :)

**Review by kantasv — COMMENTED** [review]
Hi Henri, sorry to get back to you a bit late. Here are partial reviews of mine. Could you take a look? I haven't reviewed all the diff. I'll add review meanwhile.

**Comment by henrif75** [comment]
> Hi Henri, sorry to get back to you a bit late. Here are partial reviews of mine. Could you take a look? I haven't reviewed all the diff. I'll add review meanwhile.

No worries at all, thanks for your help!

File: po/ja.po [hunk 1]
```diff
@@ -6449,217 +6763,260 @@ msgid ""
...
 
 #: src/std-traits/operators.md
 msgid ""
 "Answer: `Add:add` consumes `self`. If type `T` for which you are overloading "
 "the operator is not `Copy`, you should consider overloading the operator for "
 "`&T` as well. This avoids unnecessary cloning on the call site."
 msgstr ""
+"回答: `Add:add` は `self` を使用します。演算子をオーバーロードする型 `T` が "
+"`Copy` でない場合は、`&T` の演算子もオーバーロードすることを検討する必要があ"
+"ります。これにより、コールサイトでの不要なクローン作成を回避できます。"
```
Comment (kantasv) [inline po/ja.po:6962]:
```suggestion
"ります。これにより、呼び出し箇所での不要なクローン作成を回避できます。"
```

File: po/ja.po [hunk 2]
```diff
@@ -6449,217 +6763,260 @@ msgid ""
...
 
 #: src/std-traits/operators.md
 msgid ""
 "Why is `Output` an associated type? Could it be made a type parameter of the "
 "method?"
 msgstr ""
+"`Output` が関連する型であるのはなぜですか？これをメソッドの型パラメータにでき"
```
Comment (kantasv) [inline po/ja.po:6969]: Associated type -> 関連型 based on [Rust By Example - JA](https://doc.rust-jp.rs/rust-by-example-ja/generics/assoc_items/types.html) 
```suggestion
"`Output` が関連型であるのはなぜですか？これをメソッドの型パラメータにでき"
```

File: po/ja.po [hunk 3]
```diff
@@ -6449,217 +6763,260 @@ msgid ""
...
+"るでしょうか？"
 
 #: src/std-traits/operators.md
 msgid ""
 "Short answer: Function type parameters are controlled by the caller, but "
 "associated types (like `Output`) are controlled by the implementer of a "
 "trait."
 msgstr ""
+"短い回答: 関数型のパラメータは呼び出し元によって制御されますが、関連する型"
```
Comment (kantasv) [inline po/ja.po:6978]:
```suggestion
"短い回答: 関数型のパラメータは呼び出し元によって制御されますが、関連型"
```

File: po/ja.po [hunk 4]
```diff
@@ -6668,31 +7025,41 @@ msgid ""
 "Your function will accept types that implement `From` and those that _only_ "
 "implement `Into`."
 msgstr ""
+"「`String` に変換できるすべて」のような関数引数の入力型を宣言する場合、この"
+"ルールは逆となり、`Into`を使用する必要があります。関数は、`From` を実装する型"
+"と、`Into` のみを実装する型を受け入れます。"
```
Comment (kantasv) [inline po/ja.po:7030]:
```suggestion
"と、`Into` _のみ_ を実装する型を受け入れます。"
```

File: po/ja.po [hunk 5]
```diff
@@ -6668,31 +7025,41 @@ msgid ""
...
 
 #: src/std-traits/casting.md
 msgid ""
 "Rust has no _implicit_ type conversions, but does support explicit casts "
 "with `as`. These generally follow C semantics where those are defined."
 msgstr ""
+"Rust には暗黙的な型変換はありませんが、`as` による明示的なキャストはサポート"
```
Comment (kantasv) [inline po/ja.po:7037]:
```suggestion
"Rust には _暗黙的_ な型変換はありませんが、`as` による明示的なキャストはサポート"
```

File: po/ja.po [hunk 6]
```diff
@@ -6668,31 +7025,41 @@ msgid ""
...
-msgstr ""
+msgstr "\"as u8: {}\""
 
 #: src/std-traits/casting.md
 msgid ""
 "The results of `as` are _always_ defined in Rust and consistent across "
 "platforms. This might not match your intuition for changing sign or casting "
 "to a smaller type -- check the docs, and comment for clarity."
 msgstr ""
+"`as` の結果は Rust で常に定義され、プラットフォーム間で一貫しています。これ"
```
Comment (kantasv) [inline po/ja.po:7059]:
```suggestion
"`as` の結果は Rust で _常に_ 定義され、プラットフォーム間で一貫しています。これ"
```

File: po/ja.po [hunk 7]
```diff
@@ -6668,31 +7025,41 @@ msgid ""
...
+"は、署名を変更したり、より小さな型にキャストしたりする際に得られる直感に反し"
```
Comment (kantasv) [inline po/ja.po:7060]:
```suggestion
"は、正負の符号を変えたり、より小さな型にキャストしたりする際に得られる直感に反し"
```

File: po/ja.po [hunk 8]
```diff
@@ -6711,127 +7083,153 @@ msgid ""
 "casts, `TryFrom` and `TryInto` are available when you want to handle casts "
 "that fit differently from those that don't."
 msgstr ""
+"絶対に正しいキャスト（例: `u32` から `u64` へのキャスト）では、キャストが実際"
```
Comment (kantasv) [inline po/ja.po:7086]: @kantasv Notes to self: Reviewed until here

**Review by kantasv — COMMENTED** [review]
Hi, @henrif75  I reviewed another 30% of the diff. Could you apply changes? I'll make another review for the remaining 10% and will get back to you. Thanks!

File: po/ja.po [hunk 9]
```diff
@@ -6711,127 +7083,153 @@ msgid ""
...
+"Rust 標準ライブラリは、型が `Default` を実装できることを認識しており、それを"
+"使用するコンビニエンス メソッドを提供しています。"
 
 #: src/std-traits/default.md
 msgid ""
 "The `..` syntax is called [struct update syntax](https://doc.rust-lang.org/"
 "book/ch05-01-defining-structs.html#creating-instances-from-other-instances-"
 "with-struct-update-syntax)."
 msgstr ""
+"`..` 構文は、[構造体更新構文](https://doc.rust-lang.org/book/ch05-01-"
```
Comment (kantasv) [inline po/ja.po:7230]: Using translation from The Rust Programming Language (Japanese) https://doc.rust-jp.rs/book-ja/ch05-01-defining-structs.html

```suggestion
"`..` 構文は、[構造体更新記法](https://doc.rust-lang.org/book/ch05-01-"
```

**Review by kantasv — COMMENTED** [review]
Hi @henrif75 I have finally reached to the end of diff. Could you apply changes once again? Thanks so much for your support so far and going forward :)

**Comment by henrif75** [comment]
> Hi @henrif75 I have finally reached to the end of diff. Could you apply changes once again? Thanks so much for your support so far and going forward :)

Thank you @kantasv !

**Review by kantasv — COMMENTED** [review]
Thanks, @henrif75 for fix :) Looks good to me!

**Review by kantasv — APPROVED** [review]

**Pull request merged**